Dockerfile

```diff
@@ -1,24 +1,28 @@
-# Pull base image
-FROM python:3.12.2-slim-bookworm
+# First, build the application in the `/app` directory.
+FROM ghcr.io/astral-sh/uv:python3.12-bookworm-slim AS builder
+ENV UV_COMPILE_BYTECODE=1
+ENV UV_LINK_MODE=copy
+WORKDIR /app
+RUN --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,source=uv.lock,target=uv.lock \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    uv sync --frozen --no-install-project --no-dev
+ADD . /app
+RUN --mount=type=cache,target=/root/.cache/uv \
+    uv sync --frozen --no-dev
 
-# Set environment variables
-ENV PYTHONDONTWRITEBYTECODE 1
-ENV PYTHONUNBUFFERED 1
-
-# Create and set work directory called `app`
-RUN mkdir -p /code
-WORKDIR /code
+# Then, use a final image without uv
+FROM python:3.12-slim-bookworm
```

Comment:
Any reason to not include `uv` in our final image? It seems useful.

Reply:
Two reasons yes:
1. I think it's a best practice in most cases to avoid getting build tools in production. Your production code doesn't need it:
  - Dependencies have already been resolved and installed (that's the build step)
  - There is no multiple-Python or multiple-venv shenanigans within the container
2. [The official guide](https://github.com/astral-sh/uv-docker-example/) for using `uv` with `Docker` provided a ready-to-go multistage `Dockerfile`, so it seems not only they agree with `1.`, but also they make it trivial to implement.

Some figures:
- `python:3.12-slim-bookworm` is 124MB
- `ghcr.io/astral-sh/uv:python3.12-bookworm-slim` is 157MB

@jefftriplett do you think there are use cases for having `uv` in the container?

Reply:
It's all trade-offs, and I think that's fine. 

It might be something to re-think (maybe add a production layer) once we merge requirements.in into `pyproject.toml`. That makes having dev requirements easier. 

I kind of view these projects as more dx/dev-friendly so being inside of compose should expose uv so we can run `uv lock` or `uv add {package}` but I'm honestly still new to the Litium. 

All good and thank you for the explanation.